modules/fb_instant_articles_rss/fb_instant_articles_rss.module

```diff
@@ -178,49 +178,42 @@ function fb_instant_articles_rss_form_node_form_alter(&$form, &$form_state, $for
-    if (!empty($form['#node']->nid)) {
-      $fb_instant_enabled = fb_instant_articles_rss_is_article($form['#node']->nid);
-    }
-    else {
-      $fb_instant_enabled = fb_instant_articles_display_is_article_type('node', $type);
-    }
+    $is_type = fb_instant_articles_display_is_article_type('node', $form['#node']->type);
 
-    $previously_checked = (isset($form_state['values']) && $form_state['values']['fb_instant_articles_rss']['fb_instant_enabled']);
+    if ($is_type) {
+      // Is Enabled?
+      if (!empty($form['#node']->nid)) {
+        $fb_instant_enabled = fb_instant_articles_rss_is_article($form['#node']->nid);
+      }
 
-    $form['fb_instant_articles_rss']['fb_instant_enabled'] = array(
-      '#type' => 'checkbox',
-      '#title' => t('Include in Facebook Instant Articles RSS feed.'),
-      '#description' => t('Enable this entity to be included in the Facebook Instant Articles RSS feed.'),
-      '#weight' => 0,
-      '#default_value' => ($previously_checked || $fb_instant_enabled) ? TRUE : FALSE,
-    );
+      $form['fb_instant_articles_rss']['fb_instant_enabled'] = array(
+        '#type' => 'checkbox',
+        '#title' => t('Include in Facebook Instant Articles RSS feed.'),
+        '#description' => t('Enable this entity to be included in the Facebook Instant Articles RSS feed.'),
+        '#weight' => 0,
+        '#default_value' => ($fb_instant_enabled) ? TRUE : FALSE,
+      );
 
-    // Add custom submit.
-    $form['#submit'][] = 'fb_instant_articles_rss_node_form_submit';
+      // Add custom submit.
+      $form['#submit'][] = 'fb_instant_articles_rss_node_form_submit';
+    }
   }
 }
 
 /**
  * Submit callback for node form.
  */
 function fb_instant_articles_rss_node_form_submit($form, &$form_state) {
-  $fb_instant_enabled = fb_instant_articles_display_is_article_type('node', $form['type']['#value']);
-  if ($fb_instant_enabled) {
-    if (!isset($form_state['values']['fb_instant_enabled'])) {
-      $form_state['values']['fb_instant_enabled'] = 0;
-    }
-    $value = $form_state['values']['fb_instant_enabled'];
-    $default_enabled = variable_get('fb_instant_articles_rss_all_enabled_default', TRUE);
-    if ($default_enabled xor $value) {
-      // Save the new article entity setting if the value is different from the
-      // default value.
-      fb_instant_articles_rss_set_entity($form['nid']['#value'], $value);
-    }
-    else {
-      // Otherwise if the value and default value are the same, delete the entry
-      // to keep the table clean.
-      fb_instant_articles_rss_delete_entity($form['nid']['#value']);
-    }
+  if (!isset($form_state['values']['fb_instant_enabled'])) {
+    $form_state['values']['fb_instant_enabled'] = 0;
+  }
+  $value = $form_state['values']['fb_instant_enabled'];
+  if ($value) {
+    // Save the new article entity setting
+    fb_instant_articles_rss_set_entity($form['nid']['#value'], $value);
+  }
+  else {
+    // delete the entry to keep the table clean.
```

Comment:
Capitalize "delete".

Reply:
Wouldn't we need to store a value of 0 to explicitly say that this should be disabled? In case the default value changed from 0 to 1?